openage/convert/export/formats/sprite_metadata.py

```diff
@@ -4,52 +4,71 @@
 Sprite definition file.
 """
 
+from enum import Enum
+
 from ..data_definition import DataDefinition
 
-FILE_VERSION = "0.1.0"
+FILE_VERSION = '0.1.0'
+
+
+class LayerMode(Enum):
+    """
+    Possible values for the mode of a layer.
+    """
+    OFF = 'off'
+    ONCE = 'once'
+    LOOP = 'loop'
+
+
+class LayerPosition(Enum):
+    """
+    Possible values for the position of a layer.
+    """
+    DEFAULT = 'default'
+    ROOF = 'roof'
+    SHADOW = 'shadow'
 
 
 class SpriteMetadata(DataDefinition):
     """
     Collects sprite metadata and can format it
     as a .sprite custom format
     """
-    def __init__(self, targetdir, filename, sprite_name):
+    def __init__(self, targetdir, filename):
         super().__init__(targetdir, filename)
 
-        self.name = sprite_name
         self.image_files = {}
         self.layers = {}
         self.angles = {}
         self.frames = []
 
-    def add_image(self, id, file_name):
+    def add_image(self, img_id, filename):
         """
         Add an image and the relative file name.
 
-        :param id: Image identifier.
-        :type id: int
-        :param file_name: Name of the image file, without extension.
-        :type file_name: str
+        :param img_id: Image identifier.
+        :type img_id: int
+        :param filename: Name of the image file, with extension.
+        :type filename: str
         """
-        self.image_files[id] = file_name
+        self.image_files[img_id] = filename
 
-    def add_layer(self, id, mode, position, time_per_frame=None, replay_delay=None):
+    def add_layer(self, layer_id, mode, position, time_per_frame=None, replay_delay=None):
         """
         Add a layer and its relative parameters.
 
-        :param id: Layer identifier.
-        :type id: int
+        :param layer_id: Layer identifier.
+        :type layer_id: int
         :param mode: Animation mode (off, once, loop).
-        :type mode: str
+        :type mode: LayerMode
         :param position: Layer position (default, roof, shadow).
-        :type position: str
+        :type position: LayerPosition
```

Comment:
`int` is also an allowed type

Reply:
I don't know what do you mean. Should I just put 2 type lines to allow 2 types? And how does this fare in `dump` where I call `LayerPosition.value`, if an `int` was given instead?

Reply:
These are Sphinx docstrings ([see here](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html)), so you can assign several types seperated by commas, e.g.

```
...
:type position: int, LayerPosition
...
```

in `dump`, you have to do a type check with `isinstance` and then do something different, depending on the type.

Reply:
But would you then just dump the integer value? Or is that supposed to be converted to `default`, `roof` or `shadow`?

Reply:
You dump the int value if it is an `int` or the string `LayerPosition.value` if it is a `LayerPosition`. Internally, these strings are mapped to integers as well, they are mostly there because writing `shadow` is more user-friendly than an arbitrary integer.